compiler/cgen.nim

```diff
@@ -566,6 +566,9 @@ proc localVarDecl(p: BProc; n: PNode): Rope =
     if s.kind == skLet: incl(s.loc.flags, lfNoDeepCopy)
   if s.kind in {skLet, skVar, skField, skForVar} and s.alignment > 0:
     result.addf("NIM_ALIGN($1) ", [rope(s.alignment)])
+
+  genCLineDir(result, toFullPath(p.config, n.info), n.info.safeLineNm, p.config)
```

Comment:
`toFullPath(p.config, n.info)` is pretty expensive though and not lazily evaluated.

Reply:
I thought it would be okay because there usually aren't that many locals. Plus `genLineDir` is sprinkled everywhere and uses `toFullPath`. 

Perhaps I should use `proc genCLineDir(r: var Rope, info: TLineInfo; conf: ConfigRef)` and modify it to be:

```nim
proc genCLineDir(r: var Rope, info: TLineInfo; conf: ConfigRef) =
  if optLineDir in conf.options:
    genCLineDir(r, toFullPath(conf, info), info.safeLineNm, conf)
```

Would that be better? It does duplicate an if statement, but that is surely much less expensive than the `toFullPath` being sprinkled everywhere.

Reply:
Yes, it would be better.